packages/core/src/rest/Soundcloud.ts

```diff
@@ -1,13 +1,52 @@
+import SoundCloud from 'soundcloud-scraper';
 const apiUrl = 'https://api.soundcloud.com';
 
+let client;
+
+const generateClient = async () => {
+  const key =  await SoundCloud.Util.keygen();
+  client = new SoundCloud.Client(key);
+};
+
+generateClient();
+
 function prepareUrl(url: string): string {
   return `${url}&client_id=${process.env.SOUNDCLOUD_API_KEY}`;
 }
 
-export function soundcloudSearch(terms: string): Promise<Response> {
-  return fetch(prepareUrl(apiUrl + '/tracks?limit=50&q=' + terms));
+export async function soundcloudSearch(terms: string) {
+  const trackData = [];
+  const searchResults = await client.search(terms)
+    .then(async song => song)
+    .catch(console.error);
```

Comment:
To get the stream url we have to make 3 network requests. 

1. Search soundcloud for the correct artist name and song title (returns multiple)
2. Get the track details (returns 1)
3. Get the stream details (returns 1)